nest/src/vault/NestAtomicQueue.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { SafeCast } from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+
+import { Auth, Authority } from "@solmate/auth/Auth.sol";
+import { ERC20 } from "@solmate/tokens/ERC20.sol";
+import { FixedPointMathLib } from "@solmate/utils/FixedPointMathLib.sol";
+
+import { AtomicQueue } from "@boringvault/src/atomic-queue/AtomicQueue.sol";
+
+import { IAtomicQueue } from "../interfaces/IAtomicQueue.sol";
+import { IComponentToken } from "../interfaces/IComponentToken.sol";
+import { NestBoringVaultModule } from "./NestBoringVaultModule.sol";
+
+/**
+ * @title NestAtomicQueue
+ * @notice AtomicQueue implementation for the Nest vault
+ * @dev An AtomicQueue that only allows withdraws into a single `asset` that is
+ * configured.
+ */
+contract NestAtomicQueue is NestBoringVaultModule, AtomicQueue {
+
+    using SafeCast for uint256;
+    using FixedPointMathLib for uint256;
+
+    // Public State
+
+    // Constants
+    uint256 public constant REQUEST_ID = 0;
+
+    // Public State
+    uint256 public deadlinePeriod;
+    uint256 public pricePercentage; // Must be 4 decimals i.e. 9999 = 99.99%
```

Comment:
I'd recommend setting:

```
uint256 public constant _BASE = 1e18;
```

Then later:

```
uint88(convertToAssets(10 ** this.decimals).mulDivDown(priceMultiplier, _BASE))
```

Instead of a magic constant 10_000